references/optical_flow/transforms.py

```diff
@@ -7,16 +7,21 @@ class ValidateModelInput(torch.nn.Module):
     # Pass-through transform that checks the shape and dtypes to make sure the model gets what it expects
     def forward(self, img1, img2, flow, valid_flow_mask):
 
-        assert all(isinstance(arg, torch.Tensor) for arg in (img1, img2, flow, valid_flow_mask) if arg is not None)
-        assert all(arg.dtype == torch.float32 for arg in (img1, img2, flow) if arg is not None)
+        if not all(isinstance(arg, torch.Tensor) for arg in (img1, img2, flow, valid_flow_mask) if arg is not None):
+            raise TypeError("This method expects all input arguments to be of type torch.Tensor.")
+        if not all(arg.dtype == torch.float32 for arg in (img1, img2, flow) if arg is not None):
+            raise TypeError("This method expects the tensors img1, img2 and flow of be of dtype torch.float32.")
 
-        assert img1.shape == img2.shape
+        if not img1.shape == img2.shape:
+            raise ValueError("img1 and img2 should have the same shape.")
         h, w = img1.shape[-2:]
-        if flow is not None:
-            assert flow.shape == (2, h, w)
+        if flow is not None and not flow.shape == (2, h, w):
```

Comment:
I think the conversion from `assert` to `if not` can feel a bit unnatural in some places. E.g. here it would be more natural to check for inequality rather than have `if not a == b`. There are other places with similar quirks.

```suggestion
        if flow is not None and flow.shape != (2, h, w):
```